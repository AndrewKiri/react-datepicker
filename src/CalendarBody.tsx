import * as React from 'react';
import * as moment from 'moment';
import { getDayMatrix, isDayEqual } from './util/DateUtil';
import Day from './Day';

interface Props{
  current: moment.Moment
  selected?: moment.Moment[]
  startDay?: moment.Moment
  endDay?: moment.Moment
  onChange?: (date: moment.Moment) => void
  customDayClass?: (date: moment.Moment) => string | string[]
  customDayText?: (date: moment.Moment) => string
}
class CalendarBody extends React.Component<Props> {
  
  constructor(props:Props){
    super(props);
  }
  getCustomClass = (date: string):string => {
    const { current, customDayClass } = this.props;
    const currentDate = moment(current).date(parseInt(date));
    let classArr:string[] = [];
    
    if(!date.trim()) return '';

    classArr.push(`calendar__day--${currentDate.day()}`);

    if( customDayClass !== undefined ) {
      const customClass = customDayClass(currentDate);
      classArr = classArr.concat(typeof(customClass) === 'string' ? 
        [customClass]: customClass);
    }

    return classArr.join(' '); 
  }

  getCustomText = (date: string):string => {
    const { current, customDayText, startDay, endDay} = this.props;
    const currentDate = moment(current).date(parseInt(date));
    if(!date.trim()) return '';
    if(!customDayText) return '';

    return customDayText(currentDate);
  }

  isSelected = (date: string):boolean => {
    const { selected } = this.props;
    if(selected === undefined) return false;
    return selected.some(v => this.isDayEqual(date, v));
  }

  isDayEqual = (date: string, compareDate?: moment.Moment):boolean => {
    const { current } = this.props;
    const currentDate = moment(current).date(parseInt(date));
    if(!date.trim()) return false;
    if(!compareDate) return false;
    return isDayEqual(compareDate, currentDate);
  } 

  isRange = (date: string):boolean => {
    const { current, startDay, endDay } = this.props;
    const currentDate = moment(current).date(parseInt(date));
    
    if(!date.trim()) return false;
    if((startDay && currentDate.diff(startDay, 'days') > 0 )
      && (endDay && currentDate.diff(endDay, 'days') < 0)) return true;

    return false;
  }


  handleChange = (value: string) => {
    const { onChange, current } = this.props;

    if(onChange)
      onChange(moment(current).date(parseInt(value)));
  }

  render() {
    const {
      current,
      startDay,
      endDay
    } = this.props;

    const dayMatrix = getDayMatrix(current.year(), current.month());
    const weekdays = moment.weekdaysShort();

    return (
      <div className="calendar__body">
        <table className="calendar__body--table">
          <thead>
            <tr>
              {
                weekdays.map((v,i) => <th key={i}>{v}</th>)
              }
            </tr>
          </thead> 
          <tbody>
            {
              dayMatrix.map((row,i) => (
              <tr key={i}>
                  {
                    row.map((date,j) => (
                      <Day 
                        customClass={this.getCustomClass(date)}
                        customText={this.getCustomText(date)}
                        onChange={this.handleChange}
                        start={this.isDayEqual(date, startDay)}
                        end={this.isDayEqual(date, endDay)}
                        range={this.isRange(date)}
                        today={this.isDayEqual(date, moment())}
                        selected={this.isSelected(date)}
                        value={date} 
                        key={i+j}
                      />
                    ))     
                  }
                </tr>
              ))
            }
          </tbody>
        </table>
      </div>
  )
  }
}
export default CalendarBody;