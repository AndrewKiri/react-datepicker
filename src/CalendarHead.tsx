import * as React from 'react';

interface Props{
  onPrev?: () => void,
  onNext?: () => void,
  prevIcon?: boolean,
  nextIcon?: boolean,
  title?: string
}

const defaultProps = {
  title: ''
};

const CalendarHead: React.FunctionComponent<Props> = ({
  onPrev,
  onNext,
  prevIcon,
  nextIcon,
  title
}) => {
  return (
    <div className="calendar__head">
      <div className="calendar__head--prev">
        {
          prevIcon && (
            <button 
              onClick={onPrev}
              className="calendar__head--button"
            > 
              <i className="icon-navigate_before"/>
            </button>
          )
        }
      </div>
      <h2 className="calendar__head--title">
        { title }
      </h2>
      <div className="calendar__head--next">
        {
          nextIcon && (
            <button 
              onClick={onNext}
              className="calendar__head--button"
            > 
              <i className="icon-navigate_next"/>
            </button>
          )
        }
      </div>
    </div>
  )
}

CalendarHead.defaultProps = defaultProps;

export default CalendarHead;