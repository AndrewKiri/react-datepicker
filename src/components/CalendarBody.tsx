import * as moment from 'moment';
import * as React from 'react';
import { CalendarEnums } from '../common/@enum';
import { DatePickerDefaults } from '../common/Constant';
import { getMonthMatrix, getYearMatrix } from '../utils/DateUtil';
import DayView, { Props as DayViewProps } from './DayView';
import TableCell from './TableCell';
import TableMatrixView from './TableMatrixView';

interface CalendarBodyProps {
  /** Calendar viewMode(Year, Month, Day) */
  viewMode: CalendarEnums.ViewMode;
  /** Calendar current Date */
  current: moment.Moment;
  /** DayClick Event */
  onClick: (value: string) => void;
  /** Locale to use */
  locale: string;
}
type Props = DayViewProps & CalendarBodyProps;
class CalendarBody extends React.Component<Props> {
  public static defaultProps = {
    viewMode: CalendarEnums.ViewMode.DAY,
    onClick: () => {},
    locale: DatePickerDefaults.locale,
  };

  public render() {
    const { current, onClick, locale } = this.props;
    const cell = (className: string) => (value: string, key: number) => (
      <TableCell className={className} text={value} onClick={text => onClick(text)} key={key} />
    );
    const viewMap = {
      [CalendarEnums.ViewMode.YEAR]: (
        <TableMatrixView matrix={getYearMatrix(current.year())} cell={cell('calendar__year')} />
      ),
      [CalendarEnums.ViewMode.MONTH]: (
        <TableMatrixView matrix={getMonthMatrix(locale)} cell={cell('calendar__month')} />
      ),
      [CalendarEnums.ViewMode.DAY]: <DayView {...this.props} />,
    };

    return <div className="calendar__body">{viewMap[this.props.viewMode]}</div>;
  }
}
export default CalendarBody;
