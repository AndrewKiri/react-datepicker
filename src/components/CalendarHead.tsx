import * as React from 'react';

interface Props {
  onPrev?: () => void;
  onNext?: () => void;
  onTitleClick?: () => void;
  prevIcon?: boolean;
  nextIcon?: boolean;
  title?: string;
}

const defaultProps = {
  title: '',
};

const CalendarHead: React.FunctionComponent<Props> = ({
  onPrev,
  onNext,
  prevIcon,
  nextIcon,
  title,
  onTitleClick,
}) => {
  return (
    <div className="calendar__head">
      <div className="calendar__head--prev">
        {prevIcon && (
          <button onClick={onPrev} className="calendar__head--button">
            <i className="icon-left" />
          </button>
        )}
      </div>
      <h2 className="calendar__head--title" onClick={onTitleClick}>
        {title}
      </h2>
      <div className="calendar__head--next">
        {nextIcon && (
          <button onClick={onNext} className="calendar__head--button">
            <i className="icon-right" />
          </button>
        )}
      </div>
    </div>
  );
};

CalendarHead.defaultProps = defaultProps;

export default CalendarHead;
