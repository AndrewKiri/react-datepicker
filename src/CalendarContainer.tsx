import * as React from 'react';
import * as moment from 'moment';
import CalendarHead from './CalendarHead';
import CalendarBody from './CalendarBody';
import classNames from 'classnames';
import 'moment/locale/ko';

export interface InheritProps {
  headerFormat?: string
  locale?: string
  selected?: moment.Moment[]
  startDay?: moment.Moment
  endDay?: moment.Moment
  current: moment.Moment
  onChange?: (date: moment.Moment) => void
  customDayClass?: (date: moment.Moment) => string | string[]
  customDayText?: (date: moment.Moment) => string
  show?: boolean
  prevIcon?: boolean
  nextIcon?: boolean
}

interface PrivateProps {
  onPrev?: () => void
  onNext?: () => void
}

type Props = InheritProps & PrivateProps

class CalendarContainer extends React.Component<Props>{

  public static defaultProps = {
    headerFormat: "YYYY년 MM월",
    show: true,
    current: moment()
  }

  constructor(props: Props){
    super(props);
    moment.locale(props.locale);
  }


  render() {
    const {
      headerFormat,
      customDayClass,
      customDayText,
      selected,
      onChange,
      startDay,
      endDay,
      prevIcon,
      nextIcon,
      onPrev,
      onNext,
      show,
      current
    } = this.props;


    const calendarClass = classNames('calendar__container', {
      'calendar--show': show
    });

    return (
      <div className={calendarClass}>
        <CalendarHead
          onPrev={onPrev} 
          onNext={onNext} 
          prevIcon={prevIcon}
          nextIcon={nextIcon}
          title={current.format(headerFormat)}
        />
        <CalendarBody
          current={current}
          selected={selected}
          startDay={startDay}
          endDay={endDay}
          onChange={onChange}
          customDayClass={customDayClass}
          customDayText={customDayText}
        />
      </div>
    )
  }
}


export default CalendarContainer;
