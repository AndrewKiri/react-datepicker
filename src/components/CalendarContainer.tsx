import * as classNames from 'classnames';
import * as moment from 'moment';
import * as React from 'react';
import { IDatePicker } from '../common/@types';
import CalendarBody from './CalendarBody';
import CalendarHead from './CalendarHead';
import { Props as DayViewProps } from './DayView';
import TodayPanel from './TodayPanel';
import { ifExistCall } from '../utils/FunctionUtil';
import { DatePickerDefaults } from '../common/Constant';

interface CalendarContainerProps {
  /** Locale to use */
  locale?: string;
  /** Calendar Show or Hide */
  show?: boolean;
  /** PrevIcon Show or Hide */
  prevIcon?: boolean;
  /** NextIcon Show or Hide */
  nextIcon?: boolean;
  /** Event for Calendar day click */
  onChange?: (date: moment.Moment) => void;
  /** TodayPanel show or hide */
  showToday?: boolean;
}

interface PrivateProps {
  /** CalendarContainer base prop */
  current: moment.Moment;
  /** Default Date parameter in calendar, which is the parent component */
  base: moment.Moment;
  /** Number of months to show at once */
  showMonthCnt: number;
  /** Set Calendar initial Date  */
  setBase: (base: moment.Moment) => void;
}

export interface State {
  viewMode: IDatePicker.ViewMode;
}

export type InheritProps = DayViewProps & CalendarContainerProps;
export type Props = CalendarContainerProps & DayViewProps & PrivateProps;

class CalendarContainer extends React.Component<Props, State> {
  public static defaultProps = {
    current: moment(),
    show: true,
    showMonthCnt: 1,
    showToday: false,
    locale: DatePickerDefaults.locale,
  };

  public state = {
    viewMode: IDatePicker.ViewMode.DAY,
  };

  constructor(props: Props) {
    super(props);
    if (props.locale) {
      require(`moment/locale/${props.locale}`);
    }
  }

  public getHeaderTitle = () => {
    const { current } = this.props;
    const year = current.year();
    return {
      [IDatePicker.ViewMode.YEAR]: `${year - 4} - ${year + 5}`,
      [IDatePicker.ViewMode.MONTH]: `${year}`,
      [IDatePicker.ViewMode.DAY]: current.format('YYYY.MM'),
    }[this.state.viewMode];
  };

  public handleTitleClick = () => {
    const { viewMode } = this.state;
    const { showMonthCnt } = this.props;
    let changedMode: IDatePicker.ViewMode = viewMode;

    if (viewMode === IDatePicker.ViewMode.MONTH) {
      changedMode = IDatePicker.ViewMode.YEAR;
    } else if (viewMode === IDatePicker.ViewMode.DAY) {
      changedMode = IDatePicker.ViewMode.MONTH;
    }
    this.setState({
      viewMode: showMonthCnt > 1 ? IDatePicker.ViewMode.DAY : changedMode,
    });
  };

  public handleChange = (value: string) => {
    const { viewMode } = this.state;
    const { current, onChange, setBase, showMonthCnt, base } = this.props;
    if (!value.trim()) return;
    if (showMonthCnt > 1) {
      const date = current.date(parseInt(value, 10));
      ifExistCall(onChange, date);
      return;
    }

    if (viewMode === IDatePicker.ViewMode.YEAR) {
      setBase(base.clone().year(parseInt(value, 10)));
      this.setState({
        viewMode: IDatePicker.ViewMode.MONTH,
      });
    } else if (viewMode === IDatePicker.ViewMode.MONTH) {
      const month = moment()
        .month(value)
        .format('M');
      setBase(base.clone().month(parseInt(month, 10) - 1));
      this.setState({
        viewMode: IDatePicker.ViewMode.DAY,
      });
    } else {
      const date = current.date(parseInt(value, 10));
      ifExistCall(onChange, date);
    }
  };

  public handleBase = (method: string) => () => {
    const { base, setBase } = this.props;
    const { viewMode } = this.state;
    const clone = base.clone();
    if (viewMode === IDatePicker.ViewMode.YEAR) {
      setBase(clone[method](10, 'years'));
    } else if (viewMode === IDatePicker.ViewMode.MONTH) {
      setBase(clone[method](1, 'years'));
    } else {
      setBase(clone[method](1, 'months'));
    }
  };

  public handleToday = () => {
    const { setBase } = this.props;
    setBase(moment());
  };

  public render() {
    const {
      customDayClass,
      customDayText,
      disableDay,
      selected,
      showToday,
      startDay,
      endDay,
      prevIcon,
      nextIcon,
      show,
      onMouseOver,
      current,
      locale = DatePickerDefaults.locale,
    } = this.props;

    const calendarClass = classNames('calendar__container', {
      'calendar--show': show,
    });

    return (
      <div className={calendarClass}>
        <CalendarHead
          onPrev={this.handleBase('subtract')}
          onNext={this.handleBase('add')}
          prevIcon={prevIcon}
          nextIcon={nextIcon}
          onTitleClick={this.handleTitleClick}
          title={this.getHeaderTitle()}
        />
        {showToday && (
          <TodayPanel
            today={moment()
              .locale(locale)
              .format('LL')}
            onClick={this.handleToday}
            show={showToday}
          />
        )}
        <CalendarBody
          viewMode={this.state.viewMode}
          current={current}
          selected={selected}
          startDay={startDay}
          endDay={endDay}
          disableDay={disableDay}
          onClick={this.handleChange}
          onMouseOver={onMouseOver}
          customDayClass={customDayClass}
          customDayText={customDayText}
          locale={locale}
        />
      </div>
    );
  }
}

export default CalendarContainer;
