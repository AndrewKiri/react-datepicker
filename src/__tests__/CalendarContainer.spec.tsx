import * as React from 'react';
import * as sinon from 'sinon';
import * as moment from 'moment';
import { shallow, mount } from 'enzyme';
import CalendarContainer from '../CalendarContainer';
import CalendarHead from '../CalendarHead';
import CalendarBody from '../CalendarBody';

describe('<CalendarContainer/>', () => {
  
  it('renders with no props', () => {
    const component = shallow(
      <CalendarContainer />
    );
    
    expect(component).toMatchSnapshot();
    expect(component.find(CalendarHead)).toHaveLength(1);
    expect(component.find(CalendarBody)).toHaveLength(1);
  });

  it('props show correctly', () => {
    const component = shallow(
      <CalendarContainer 
        show
      />
    );
    
    expect(component).toMatchSnapshot();
    expect(component.hasClass('calendar--show')).toBeTruthy();
  });

  it('props headerFormat correctly', () => {
    const component = shallow(
      <CalendarContainer 
        show
        current={moment('20190101')}
        headerFormat="YYYY/MM/DD"
      />
    );
    
    expect(component).toMatchSnapshot();
    expect(component.find(CalendarHead).props().title).toEqual('2019/01/01');
  });

  it('props locale correctly', () => {
    const en = mount(
      <CalendarContainer 
        show
        current={moment('20190101')}
        locale="en"
      />
    );
    
    const ko = mount(
      <CalendarContainer 
        show
        current={moment('20190101')}
        locale="ko"
      />
    );
    expect(en).toMatchSnapshot();
    expect(en.find('th').first().text()).toEqual('Sun');
    expect(ko.find('th').first().text()).toEqual('Ïùº');
  });
});