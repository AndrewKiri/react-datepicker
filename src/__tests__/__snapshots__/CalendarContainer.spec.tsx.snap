// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CalendarContainer/> props headerFormat correctly 1`] = `
<div
  className="calendar__container calendar--show"
>
  <calendarHead
    title="2018/12/01"
  />
  <CalendarBody
    current={"2018-12-01T00:00:00.000Z"}
  />
</div>
`;

exports[`<CalendarContainer/> props locale correctly 1`] = `
<CalendarContainer
  current={"2018-12-01T00:00:00.000Z"}
  headerFormat="YYYY년 MM월"
  locale="en"
  show={true}
>
  <div
    className="calendar__container calendar--show"
  >
    <calendarHead
      title="2018년 12월"
    >
      <div
        className="calendar__head"
      >
        <div
          className="calendar__head--prev"
        />
        <h2
          className="calendar__head--title"
        >
          2018년 12월
        </h2>
        <div
          className="calendar__head--next"
        />
      </div>
    </calendarHead>
    <CalendarBody
      current={"2018-12-01T00:00:00.000Z"}
    >
      <div
        className="calendar__body"
      >
        <table
          className="calendar__body--table"
        >
          <thead>
            <tr>
              <th
                key="0"
              >
                Sun
              </th>
              <th
                key="1"
              >
                Mon
              </th>
              <th
                key="2"
              >
                Tue
              </th>
              <th
                key="3"
              >
                Wed
              </th>
              <th
                key="4"
              >
                Thu
              </th>
              <th
                key="5"
              >
                Fri
              </th>
              <th
                key="6"
              >
                Sat
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              key="0"
            >
              <Day
                customClass=""
                customText=""
                end={false}
                key="0"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value=" "
              >
                <td
                  className="calendar__day"
                  onClick={[Function]}
                >
                  <span>
                     
                  </span>
                </td>
              </Day>
              <Day
                customClass=""
                customText=""
                end={false}
                key="1"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value=" "
              >
                <td
                  className="calendar__day"
                  onClick={[Function]}
                >
                  <span>
                     
                  </span>
                </td>
              </Day>
              <Day
                customClass=""
                customText=""
                end={false}
                key="2"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value=" "
              >
                <td
                  className="calendar__day"
                  onClick={[Function]}
                >
                  <span>
                     
                  </span>
                </td>
              </Day>
              <Day
                customClass=""
                customText=""
                end={false}
                key="3"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value=" "
              >
                <td
                  className="calendar__day"
                  onClick={[Function]}
                >
                  <span>
                     
                  </span>
                </td>
              </Day>
              <Day
                customClass=""
                customText=""
                end={false}
                key="4"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value=" "
              >
                <td
                  className="calendar__day"
                  onClick={[Function]}
                >
                  <span>
                     
                  </span>
                </td>
              </Day>
              <Day
                customClass=""
                customText=""
                end={false}
                key="5"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value=" "
              >
                <td
                  className="calendar__day"
                  onClick={[Function]}
                >
                  <span>
                     
                  </span>
                </td>
              </Day>
              <Day
                customClass="calendar__day--6"
                customText=""
                end={false}
                key="6"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value="1"
              >
                <td
                  className="calendar__day calendar__day--6"
                  onClick={[Function]}
                >
                  <span>
                    1
                  </span>
                </td>
              </Day>
            </tr>
            <tr
              key="1"
            >
              <Day
                customClass="calendar__day--0"
                customText=""
                end={false}
                key="1"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value="2"
              >
                <td
                  className="calendar__day calendar__day--0"
                  onClick={[Function]}
                >
                  <span>
                    2
                  </span>
                </td>
              </Day>
              <Day
                customClass="calendar__day--1"
                customText=""
                end={false}
                key="2"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value="3"
              >
                <td
                  className="calendar__day calendar__day--1"
                  onClick={[Function]}
                >
                  <span>
                    3
                  </span>
                </td>
              </Day>
              <Day
                customClass="calendar__day--2"
                customText=""
                end={false}
                key="3"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value="4"
              >
                <td
                  className="calendar__day calendar__day--2"
                  onClick={[Function]}
                >
                  <span>
                    4
                  </span>
                </td>
              </Day>
              <Day
                customClass="calendar__day--3"
                customText=""
                end={false}
                key="4"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value="5"
              >
                <td
                  className="calendar__day calendar__day--3"
                  onClick={[Function]}
                >
                  <span>
                    5
                  </span>
                </td>
              </Day>
              <Day
                customClass="calendar__day--4"
                customText=""
                end={false}
                key="5"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value="6"
              >
                <td
                  className="calendar__day calendar__day--4"
                  onClick={[Function]}
                >
                  <span>
                    6
                  </span>
                </td>
              </Day>
              <Day
                customClass="calendar__day--5"
                customText=""
                end={false}
                key="6"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value="7"
              >
                <td
                  className="calendar__day calendar__day--5"
                  onClick={[Function]}
                >
                  <span>
                    7
                  </span>
                </td>
              </Day>
              <Day
                customClass="calendar__day--6"
                customText=""
                end={false}
                key="7"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value="8"
              >
                <td
                  className="calendar__day calendar__day--6"
                  onClick={[Function]}
                >
                  <span>
                    8
                  </span>
                </td>
              </Day>
            </tr>
            <tr
              key="2"
            >
              <Day
                customClass="calendar__day--0"
                customText=""
                end={false}
                key="2"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value="9"
              >
                <td
                  className="calendar__day calendar__day--0"
                  onClick={[Function]}
                >
                  <span>
                    9
                  </span>
                </td>
              </Day>
              <Day
                customClass="calendar__day--1"
                customText=""
                end={false}
                key="3"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value="10"
              >
                <td
                  className="calendar__day calendar__day--1"
                  onClick={[Function]}
                >
                  <span>
                    10
                  </span>
                </td>
              </Day>
              <Day
                customClass="calendar__day--2"
                customText=""
                end={false}
                key="4"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value="11"
              >
                <td
                  className="calendar__day calendar__day--2"
                  onClick={[Function]}
                >
                  <span>
                    11
                  </span>
                </td>
              </Day>
              <Day
                customClass="calendar__day--3"
                customText=""
                end={false}
                key="5"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value="12"
              >
                <td
                  className="calendar__day calendar__day--3"
                  onClick={[Function]}
                >
                  <span>
                    12
                  </span>
                </td>
              </Day>
              <Day
                customClass="calendar__day--4"
                customText=""
                end={false}
                key="6"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value="13"
              >
                <td
                  className="calendar__day calendar__day--4"
                  onClick={[Function]}
                >
                  <span>
                    13
                  </span>
                </td>
              </Day>
              <Day
                customClass="calendar__day--5"
                customText=""
                end={false}
                key="7"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value="14"
              >
                <td
                  className="calendar__day calendar__day--5"
                  onClick={[Function]}
                >
                  <span>
                    14
                  </span>
                </td>
              </Day>
              <Day
                customClass="calendar__day--6"
                customText=""
                end={false}
                key="8"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value="15"
              >
                <td
                  className="calendar__day calendar__day--6"
                  onClick={[Function]}
                >
                  <span>
                    15
                  </span>
                </td>
              </Day>
            </tr>
            <tr
              key="3"
            >
              <Day
                customClass="calendar__day--0"
                customText=""
                end={false}
                key="3"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value="16"
              >
                <td
                  className="calendar__day calendar__day--0"
                  onClick={[Function]}
                >
                  <span>
                    16
                  </span>
                </td>
              </Day>
              <Day
                customClass="calendar__day--1"
                customText=""
                end={false}
                key="4"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value="17"
              >
                <td
                  className="calendar__day calendar__day--1"
                  onClick={[Function]}
                >
                  <span>
                    17
                  </span>
                </td>
              </Day>
              <Day
                customClass="calendar__day--2"
                customText=""
                end={false}
                key="5"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value="18"
              >
                <td
                  className="calendar__day calendar__day--2"
                  onClick={[Function]}
                >
                  <span>
                    18
                  </span>
                </td>
              </Day>
              <Day
                customClass="calendar__day--3"
                customText=""
                end={false}
                key="6"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value="19"
              >
                <td
                  className="calendar__day calendar__day--3"
                  onClick={[Function]}
                >
                  <span>
                    19
                  </span>
                </td>
              </Day>
              <Day
                customClass="calendar__day--4"
                customText=""
                end={false}
                key="7"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value="20"
              >
                <td
                  className="calendar__day calendar__day--4"
                  onClick={[Function]}
                >
                  <span>
                    20
                  </span>
                </td>
              </Day>
              <Day
                customClass="calendar__day--5"
                customText=""
                end={false}
                key="8"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value="21"
              >
                <td
                  className="calendar__day calendar__day--5"
                  onClick={[Function]}
                >
                  <span>
                    21
                  </span>
                </td>
              </Day>
              <Day
                customClass="calendar__day--6"
                customText=""
                end={false}
                key="9"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value="22"
              >
                <td
                  className="calendar__day calendar__day--6"
                  onClick={[Function]}
                >
                  <span>
                    22
                  </span>
                </td>
              </Day>
            </tr>
            <tr
              key="4"
            >
              <Day
                customClass="calendar__day--0"
                customText=""
                end={false}
                key="4"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value="23"
              >
                <td
                  className="calendar__day calendar__day--0"
                  onClick={[Function]}
                >
                  <span>
                    23
                  </span>
                </td>
              </Day>
              <Day
                customClass="calendar__day--1"
                customText=""
                end={false}
                key="5"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value="24"
              >
                <td
                  className="calendar__day calendar__day--1"
                  onClick={[Function]}
                >
                  <span>
                    24
                  </span>
                </td>
              </Day>
              <Day
                customClass="calendar__day--2"
                customText=""
                end={false}
                key="6"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value="25"
              >
                <td
                  className="calendar__day calendar__day--2"
                  onClick={[Function]}
                >
                  <span>
                    25
                  </span>
                </td>
              </Day>
              <Day
                customClass="calendar__day--3"
                customText=""
                end={false}
                key="7"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value="26"
              >
                <td
                  className="calendar__day calendar__day--3"
                  onClick={[Function]}
                >
                  <span>
                    26
                  </span>
                </td>
              </Day>
              <Day
                customClass="calendar__day--4"
                customText=""
                end={false}
                key="8"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value="27"
              >
                <td
                  className="calendar__day calendar__day--4"
                  onClick={[Function]}
                >
                  <span>
                    27
                  </span>
                </td>
              </Day>
              <Day
                customClass="calendar__day--5"
                customText=""
                end={false}
                key="9"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value="28"
              >
                <td
                  className="calendar__day calendar__day--5"
                  onClick={[Function]}
                >
                  <span>
                    28
                  </span>
                </td>
              </Day>
              <Day
                customClass="calendar__day--6"
                customText=""
                end={false}
                key="10"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value="29"
              >
                <td
                  className="calendar__day calendar__day--6"
                  onClick={[Function]}
                >
                  <span>
                    29
                  </span>
                </td>
              </Day>
            </tr>
            <tr
              key="5"
            >
              <Day
                customClass="calendar__day--0"
                customText=""
                end={false}
                key="5"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value="30"
              >
                <td
                  className="calendar__day calendar__day--0"
                  onClick={[Function]}
                >
                  <span>
                    30
                  </span>
                </td>
              </Day>
              <Day
                customClass="calendar__day--1"
                customText=""
                end={false}
                key="6"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value="31"
              >
                <td
                  className="calendar__day calendar__day--1"
                  onClick={[Function]}
                >
                  <span>
                    31
                  </span>
                </td>
              </Day>
              <Day
                customClass=""
                customText=""
                end={false}
                key="7"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value=" "
              >
                <td
                  className="calendar__day"
                  onClick={[Function]}
                >
                  <span>
                     
                  </span>
                </td>
              </Day>
              <Day
                customClass=""
                customText=""
                end={false}
                key="8"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value=" "
              >
                <td
                  className="calendar__day"
                  onClick={[Function]}
                >
                  <span>
                     
                  </span>
                </td>
              </Day>
              <Day
                customClass=""
                customText=""
                end={false}
                key="9"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value=" "
              >
                <td
                  className="calendar__day"
                  onClick={[Function]}
                >
                  <span>
                     
                  </span>
                </td>
              </Day>
              <Day
                customClass=""
                customText=""
                end={false}
                key="10"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value=" "
              >
                <td
                  className="calendar__day"
                  onClick={[Function]}
                >
                  <span>
                     
                  </span>
                </td>
              </Day>
              <Day
                customClass=""
                customText=""
                end={false}
                key="11"
                onChange={[Function]}
                range={false}
                selected={false}
                start={false}
                today={false}
                value=" "
              >
                <td
                  className="calendar__day"
                  onClick={[Function]}
                >
                  <span>
                     
                  </span>
                </td>
              </Day>
            </tr>
          </tbody>
        </table>
      </div>
    </CalendarBody>
  </div>
</CalendarContainer>
`;

exports[`<CalendarContainer/> props show correctly 1`] = `
<div
  className="calendar__container calendar--show"
>
  <calendarHead
    title="2018년 12월"
  />
  <CalendarBody
    current={"2018-12-01T00:00:00.000Z"}
  />
</div>
`;

exports[`<CalendarContainer/> renders with no props 1`] = `
<div
  className="calendar__container calendar--show"
>
  <calendarHead
    title="2018년 12월"
  />
  <CalendarBody
    current={"2018-12-01T00:00:00.000Z"}
  />
</div>
`;
